### PR TITLE
Fix logic leading to negative expiry

### PR DESCRIPTION
**Pull request by ghalse**
The frontpage_federation template in 1.19 displays expiry dates in the future with a negative expiry:

`European University Cyprus (expired -95.59448483 hours ago)`

This has caused more than one brief moment of panic for metadata that still has a long way to go.

And since we're here, it also uses a precision such that we're measuring microseconds.

### CODE REVIEW

**Comment by tvdijen** [comment]
Thanks @ghalse ! Would you mind patching the new frontpage in the admin module as well?

**Comment by tvdijen** [comment]
Let me rephrase that question, because maybe that one is correct...
So, could you verify that the representation in `module.php/admin/federation` is correct and doesn't yield the same bug?

**Comment by ghalse** [comment]
> Let me rephrase that question:
> Could you verify that the representation in `module.php/admin/federation` is correct and doesn't yield the same bug?

In the 1.19 branch it seems to be fine.

**Comment by tvdijen** [comment]
Thanks for verifying this.. I will merge this PR then

**Review by tvdijen — APPROVED** [review]
